torch/distributed/_spmd/api.py

```diff
@@ -306,6 +318,43 @@ def _fused_adam_decomp(
 }
 
 
+DEDUP_TARGETS: Set[torch._ops.OpOverload] = {
+    aten.all_reduce.default,
+    aten.wait_tensor.default,
+}
+
+
+def _lint(gm: fx.GraphModule) -> fx.GraphModule:
```

Comment:
I'm wondering if we should make this a separate pass instead of putting this to api.py and maybe call it sth like dedup_collectives instead of lint? We might have other collectives to dedeup soon (allgather/reduce_scatter)